packages/material-ui-lab/src/Rating/Rating.js

```diff
@@ -0,0 +1,475 @@
+        },
+        className,
+      )}
+      {...other}
+    >
+      {!readOnly && !disabled && value == null && (
+        <React.Fragment>
+          <input
+            value="0"
+            id={`${name}-0`}
+            type="radio"
+            name={name}
+            defaultChecked
+            className={classes.visuallyhidden}
+          />
+          <label htmlFor={`${name}-0`} className={classes.pristine}>
+            <span className={classes.visuallyhidden}>{getLabelText(0)}</span>
+          </label>
+        </React.Fragment>
+      )}
+      {Array.from(new Array(max)).map((_, index) => {
+        const itemValue = index + 1;
+
+        if (precision < 1) {
+          const items = Array.from(new Array(1 / precision));
+          return (
+            <div
+              key={itemValue}
+              className={clsx(classes.decimal, {
+                [classes.iconActive]:
+                  itemValue === Math.ceil(value) && (hover !== -1 || focus !== -1),
+              })}
+            >
+              {items.map(($, indexDecimal) => {
+                const itemDeciamlValue = itemValue - 1 + (indexDecimal + 1) * precision;
+
+                return item(
+                  {
+                    value: itemDeciamlValue,
+                    style:
+                      items.length - 1 === indexDecimal
+                        ? {}
+                        : {
+                            width:
+                              itemDeciamlValue === value
+                                ? `${(indexDecimal + 1) * precision * 100}%`
+                                : '0%',
+                            overflow: 'hidden',
+                            zIndex: 1,
+                            position: 'absolute',
+                          },
+                  },
+                  {
+                    filled: itemDeciamlValue <= value,
+                    hover: itemDeciamlValue <= hover,
+                    focus: itemDeciamlValue <= focus,
+                    checked: itemDeciamlValue === valueProp,
+                  },
+                );
+              })}
+            </div>
+          );
+        }
+
+        return item(
+          {
+            value: itemValue,
+          },
+          {
+            active: itemValue === value && (hover !== -1 || focus !== -1),
+            filled: itemValue <= value,
+            hover: itemValue <= hover,
+            focus: itemValue <= focus,
+            checked: itemValue === Math.round(valueProp),
+          },
+        );
+      })}
+    </Component>
+  );
+});
+
+Rating.propTypes = {
+  /**
+   * Override or extend the styles applied to the component.
+   * See [CSS API](#css) below for more details.
+   */
+  classes: PropTypes.object.isRequired,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
+  /**
+   * The component used for the root node.
+   * Either a string to use a DOM element or a component.
+   */
+  component: PropTypes.elementType,
```

Comment:
We should not default to this. Unless there is a use case we shouldn't prematurely make these internals public.

Reply:
I'm not aware of any valid use case for the prop here. I have added it for consistency, as an effort to have all the components support it. However, we can revisit this long term goal. 
What's internal about this API? Should we remove `componentProp` from the describeConformance.js default tests?

Reply:
> What's internal about this API?

internal is the rendered host node. If there's no use case we should not prematurely expose those.

Reply:
OK, will remove it.